cirq/google/calibration/workflow.py

```diff
@@ -337,19 +351,150 @@ def run_characterizations(
+) -> CircuitCalibration:
+    """Compensates circuit against errors in zeta, chi and gamma angles.
+
+    This method creates a new circuit with a single-qubit Z gates added in a such way so that
+    zeta, chi and gamma angles discovered by characterizations are cancelled-out and set to 0.
+
+    This function preserves a moment structure of the circuit. All single qubit gates appear on new
+    moments in the final circuit.
+
+    Args:
+        circuit_calibration: Description of the circuit together with its calibration metadata that
+            is compatible with the characterizations argument.
+        characterizations: List of characterization results.
+        gates_translator: Function that translates a gate to a supported FSimGate which will undergo
+            characterization. Defaults to sqrt_iswap_gates_translator.
+
+    Returns:
+        Calibrated circuit together with its calibration metadata in CircuitCalibration object. The
+        calibrated circuit has single-qubit Z gates added which compensates for the true gates
+        imperfections.
+    """
+    if len(circuit_calibration.circuit) != len(circuit_calibration.moment_allocations):
+        raise ValueError('Moment allocations does not match circuit length')
+
+    default_phases = PhasedFSimCharacterization(zeta=0.0, chi=0.0, gamma=0.0)
+
+    compensated = Circuit()
+    compensated_allocations = []
+    for moment, characterization_index in zip(
+        circuit_calibration.circuit, circuit_calibration.moment_allocations
+    ):
+        parameters = None
+        if characterization_index is not None:
+            parameters = characterizations[characterization_index]
+
+        decompositions = []
+        other = []
+        new_moment_mapping = None
+        for op in moment:
+            if not isinstance(op, GateOperation):
+                raise IncompatibleMomentError(
+                    'Moment contains operation different than GateOperation'
+                )
+
+            if isinstance(op.gate, (MeasurementGate, SingleQubitGate)):
+                other.append(op)
+            else:
+                a, b = op.qubits
+
+                translated_gate = gates_translator(op.gate)
+                if translated_gate is None:
+                    raise IncompatibleMomentError(
+                        f'Moment {moment} contains unsupported non-single qubit operation {op}'
+                    )
+
+                if parameters is None:
+                    raise ValueError(f'Missing characterization data for moment {moment}')
+
+                pair_parameters = parameters.get_parameters(a, b)
+                if pair_parameters is None:
+                    raise ValueError(
+                        f'Missing characterization data for pair {(a, b)} in {parameters}'
+                    )
+                pair_parameters = pair_parameters.merge_with(default_phases)
+
+                corrected = PhaseCorrectedFSimOperations(
```

Comment:
see comment below, but I think the variable name should be `corrections`

Reply:
Done